src/rest_resources/__tests__/base.test.ts

```diff
@@ -0,0 +1,396 @@
+import {Session} from '../../auth/session';
+import {RestResourceRequestError} from '../../error';
+
+import FakeResource from './fake_resource';
+import FakeResourceWithCustomPrefix from './fake_resource_with_custom_prefix';
+
+describe('Base REST resource', () => {
```

Comment:
I think this class could benefit from reduced duplication in the different test cases.

I don't think extracting an object like in the prod code above would make sense, but helper functions could go a long way.

Reply:
I kind of like it the way it is, because the tests are essentially doing the Arrange -> Act -> Assert flow, and each of the calls there are a single call to the functions that do their part.